channels/models.py

```diff
@@ -206,8 +212,30 @@ class Article(TimestampedModel):
 
     author = models.ForeignKey(settings.AUTH_USER_MODEL, on_delete=models.CASCADE)
     post = models.OneToOneField(Post, on_delete=models.CASCADE)
+    cover_image = models.ImageField(
+        null=True, blank=True, max_length=2083, upload_to=article_image_uri
+    )
+    cover_image_small = models.ImageField(
+        null=True, blank=True, max_length=2083, upload_to=article_image_uri_small
+    )
     content = JSONField()
 
+    def save(
+        self, *args, update_image=False, **kwargs
+    ):  # pylint: disable=arguments-differ
+        if update_image:
```

Comment:
I'm wondering, could we also just do a check here to see if the update being made involves a change to the cover image, and use that fact to figure out whether we need to update the thumbnail and all? just wondering, because then I think we could, when we're updating this model, just pass in a new image or not and it would take care automatically of whether it needs to update the thumbnail or not, and we wouldn't need to know this detail of the `.save` method when we're calling it.

Reply:
This current code is consistent with how images are being saved for channels (avatars, banners) and profiles: both the `Channel` model and `Profile` model have custom `save` methods using the same logic.